Only set Authorization header when basic auth is required

kruppel opened the pull request:
As stupid as it sounds, some services will attempt to authenticate an unprotected route when an Authorization header is set.

kruppel commented:
> Coverage increased (+0.008%) to 98.484% when pulling 9d9b6af on kruppel:kurt/only-set-basic-auth-when-required into 144a8cd on sportngin:master.

Any suggestion on what to do with runs against 1.9.3 in the test matrix is more than welcome :smile:.

pbyrne commented:
@kruppel My gut reaction is to drop Ruby 1.9.3 from the build matrix. That version was end-of-lifed last year, so I feel comfortable dropping support for it. Care to make that change in this PR?

pbyrne commented on lib/ok_computer/built_in_checks/http_check.rb:
```diff
@@ -35,10 +35,16 @@ def check
-        )
+        options = { read_timeout: request_timeout }
+
+        if self.basic_auth_username || self.basic_auth_password
```
A few nit-picks on this implementation:

* These `self`s are unnecessary
* Some duplication can be eliminated if we leave the `basic_auth_options` method in-place:

```ruby
if basic_auth_options.any?
  options[:http_basic_authentication] = basic_auth_options
end
```

kruppel commented:
@pbyrne Thanks for the review! Updated based on your nits and added a commit to remove 1.9.3 from the Travis testing  matrix. Let me know if there's anything else to do.

pbyrne commented on spec/ok_computer/built_in_checks/http_check_spec.rb:
```diff
@@ -111,14 +139,5 @@ module OkComputer
       end
     end
-
-    describe '#basic_auth_options' do
```
One last nit. Could you recover this test (now that we're leaving `#basic_auth_options` in place)? With that and a green build, I'm happy to merge and ship a new version of the gem.

kruppel replied:
Whoop, ya, my bad.

pbyrne commented:
v1.7.2 is out. Thanks for contributing!
